README.md

```diff
@@ -1,3 +1,102 @@
+
+At its core, RukPak is a small set of APIs, packaged as Kubernetes CustomResourceDefinitions, and controllers that watch for those APIs. These APIs express what 
+content is being installed on-cluster and how to create a running instance of the content. 
+
+## Components 
+
+RukPak is composed of a few primary APIs: `Bundle`, `Instance`, and `Provisioner/ProvisionerClass`
+
+### Bundle
+A `Bundle` represents content that needs to be made available to other consumers in the cluster.
+Much like the contents of a container image need to be pulled and unpacked in order for Pods to start using them, 
+`Bundles` are used to reference content that may need to be pulled and should be unpacked. 
+In this sense, Bundle is a generalization of the image concept, and can be used to represent any type of content.
+
+`Bundles` do nothing on their own - they require a `Provisioner` to unpack and make their content available in-cluster. They can be
+unpacked to any arbitrary storage medium such as a PersistentVolume or ConfigMap. Each `Bundle` has an associated `spec.class` field which 
+indicates the `Provisioner` that should be watching and unpacking that particular bundle type. 
+
+Example Bundle
+```yaml
+apiVersion: core.RukPak.io/v1alpha1
+kind: Bundle
+metadata: 
+  name: example-operator.v0.9.3
+spec:
+  class: olm.bundle
+  refs:
+  - file://content
+  volumeMounts:
+  - mountPath: /content
+    configMap:
+      name: local 
+      namespace: plumbus
+  URI: my-bundle.svc.local/content
+```
+
+### Instance 
+The `Instance` API points to a Bundle and indicates that it should be “active”. This includes pivoting from older versions of an active bundle. 
+`Instance` may also include an embedded spec for a desired Bundle.
+
+Much like Pods stamp out instances of container images, `Instances` stamp out an instance of Bundles. `Instance` can be seen as a generalization of the Pod concept.
+
+The specifics of how an `Instance` makes changes to a cluster based on a referenced `Bundle` is defined by the 
+`ProvisionerClass` and the `Provisioner` that is configured to handle that `ProvisionerClass`.
+
+Example Instance
+```yaml
+apiVersion: core.RukPak.io/v1alpha1
+kind: Instance
+metadata:
+  name: resolved-654adh
+spec:
+  selector: 
+    matchLabels:
+      subscription: etcd-operator
+  bundle:
+    name: resolved-654adh
+    spec:
+      class: olm.bundle
+      refs:
+      - file://content
+      volumeMounts:
+      - mountPath: /content
+        configMap:
+          name: resolved-654adh-content
+          namespace: olm
+```
+
+### Provisioner/ProvisionerClass
+`ProvisionerClass` defines a configuration for a `Provisioner`. Provisioners are controllers that understand `Instance`, `Bundle`, and `ProvisionerClass` APIs and take action.
+
+Each `Provisioner` has a unique id. For example, the `Provisioner` that ships with OLM and understand operator bundles has the id: operators.coreos.com/operators.
```

Comment:
I think we'd like to substitute the OLM language here in favor of a rukpak-agnostic provisioner ID: https://github.com/operator-framework/rukpak/blob/main/pkg/k8s/provisioner/bundle_controller.go#L28.